azero/contracts/most/lib.rs

```diff
@@ -350,11 +364,16 @@ pub mod most {
                     amount,
                 )?;
 
+                let unclaimed_rewards_all =
+                    data.rewards_all.saturating_sub(data.paid_out_rewards_all);
+
                 // bootstrap account with pocket money
-                // NOTE: we don't revert on a failure!
-                _ = self
-                    .env()
-                    .transfer(dest_receiver_address.into(), data.pocket_money);
+                if self.env().balance() >= unclaimed_rewards_all.saturating_add(data.pocket_money) {
+                    // don't revert if the transfer fails
+                    _ = self
+                        .env()
+                        .transfer(dest_receiver_address.into(), data.pocket_money);
```

Comment:
Question: Do I understand this correctly?

1. We pay out the pocket money only if the balance of the contract is more than the rewards. 
2. So in effect we prioritize paying out the rewards over paying out pocket money.
3. The money for those two activities is mixed in the contract balance.

I'm a bit worried that this setup will make it difficult to manage the balance available for pocket money. In particular, if we have a script that says "make sure this account has at least $1000", could it be that someone does not collect their rewards and, after some time, we stop reloading the account, because the balance is too large, but all of that money is "claimed" by guardians?

Reply:
Yes, that's correct. 
I might add a helper method for fetching balance that can be distributed as pocket money, do you think it would be useful?

Reply:
alternatively we make topping up via a tx and keep track of the balance for pocket monies in it' sown storage?

Reply:
With the current fee structure it would be assumed to be neither rewards nor balance for pocket money (as we track rewards explicitly)

Reply:
yes, so you'd track rewards and pocket money explicitely, every time you send pocket money you decrease it, every time it is topped up (with a tx) you increase it. Only send out if you leave positive balance afterwards

The downside is that you cannot send A0 with a normal balances transfer this way, you need to do it with a contract tx, or else the moneyz is stuck

Reply:
So you track both rewards and pocket money? What advantage does it have over the current approach?

Reply:
Well, I'm thinking of the person who has to do the top-up - it's either going to be a script, or someone is going to do this manually. If script, then I assume (because the script doesn't exist yet) that it will just look at the account balance (because we won't remember/the person writing the script won't know about this corner case), which will cause this problem. If manually, then the person doing the top-up will need some way to check, so perhaps a contract method that exposes remaining pool of pocket money would be useful?

Reply:
I guess simple, explicit & easier to read - also easy to see when exactly to top up - one variable in storage vs having two

Reply:
OK, will rewrite it that way

Reply:
your call - I'm just suggesting it as an option (not a blocking request) . The fact you have to top up with a transaction can be seen as a potential downside